src/PermissionServiceProvider.php

```diff
@@ -124,30 +124,19 @@ public static function bladeMethodWrapper($method, $role, $guard = null): bool
 
     protected function registerBladeExtensions($bladeCompiler): void
     {
+        /** @var BladeCompiler $bladeCompiler */
         $bladeMethodWrapper = '\\Spatie\\Permission\\PermissionServiceProvider::bladeMethodWrapper';
 
-        $bladeCompiler->directive('role', fn ($args) => "<?php if({$bladeMethodWrapper}('hasRole', {$args})): ?>");
-        $bladeCompiler->directive('elserole', fn ($args) => "<?php elseif({$bladeMethodWrapper}('hasRole', {$args})): ?>");
-        $bladeCompiler->directive('endrole', fn () => '<?php endif; ?>');
+        // permission checks
+        $bladeCompiler->if('haspermission', fn ($args) => $bladeMethodWrapper('checkPermissionTo', $args));
 
-        $bladeCompiler->directive('haspermission', fn ($args) => "<?php if({$bladeMethodWrapper}('checkPermissionTo', {$args})): ?>");
-        $bladeCompiler->directive('elsehaspermission', fn ($args) => "<?php elseif({$bladeMethodWrapper}('checkPermissionTo', {$args})): ?>");
-        $bladeCompiler->directive('endhaspermission', fn () => '<?php endif; ?>');
-
-        $bladeCompiler->directive('hasrole', fn ($args) => "<?php if({$bladeMethodWrapper}('hasRole', {$args})): ?>");
-        $bladeCompiler->directive('endhasrole', fn () => '<?php endif; ?>');
-
-        $bladeCompiler->directive('hasanyrole', fn ($args) => "<?php if({$bladeMethodWrapper}('hasAnyRole', {$args})): ?>");
-        $bladeCompiler->directive('endhasanyrole', fn () => '<?php endif; ?>');
-
-        $bladeCompiler->directive('hasallroles', fn ($args) => "<?php if({$bladeMethodWrapper}('hasAllRoles', {$args})): ?>");
-        $bladeCompiler->directive('endhasallroles', fn () => '<?php endif; ?>');
-
-        $bladeCompiler->directive('unlessrole', fn ($args) => "<?php if(! {$bladeMethodWrapper}('hasRole', {$args})): ?>");
```

Comment:
`unlessrole` is missing

Reply:
I believe he's intentionally removed it because since Laravel 6.x the `Blade::if()` call auto-registers all of `if/else/unless/end` variants:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/View/Compilers/BladeCompiler.php#L701-L733

Reply:
So, does it mean that the tests should be changed?

Reply:
**PR:** https://github.com/alissn/laravel-permission/pull/1
**TESTS:** [laravel-permission/actions/runs/8103641699/job/2214866359](https://github.com/erikn69/laravel-permission/actions/runs/8103641699/job/22148663597)

Reply:
I don't see any intention to merge it. 😄😄

Reply:
Hi @parallels999, @drbyte, @erikn69,
Sorry for the delay.
I've merged the pull request.